src/java.base/share/classes/java/lang/String.java

```diff
@@ -1015,10 +1026,9 @@ private static String newStringNoRepl1(byte[] src, Charset cs) {
             throw new IllegalArgumentException(x);
         }
         if (COMPACT_STRINGS) {
-            byte[] bs = StringUTF16.compress(ca, 0, caLen);
-            if (bs != null) {
-                return new String(bs, LATIN1);
-            }
+            byte[] val = StringUTF16.compress(ca, 0, caLen);
+            int coder = StringUTF16.coderFromArrayLen(val, len);
+            return new String(val, coder);
```

Comment:
This is also missing the (conditional) call to `initCompressionFlag()`.
Worse, this uses an inappropriate constructor (`coder` will be misinterpreted as `hibyte`).

Reply:
Updated